pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/services/PluginCheckerService.kt

```diff
@@ -3,200 +3,237 @@
         }
-        if (!groovyPlugins.pitestPluginAvailable) {
-            groovyPlugins.pitestPluginAvailable = groovyChecker.pitestPluginAvailable
+    }
+
+    private fun checkForCompanionPlugin(connection: ProjectConnection) {
+        try {
+            val project = connection.getModel(GradleProject::class.java)
+            // Recursively search for the task in the project and its subprojects
+            isCompanionPluginAvailable = findTaskRecursively(project)
+        } catch (e: Exception) {
+            logger.warn("Error while checking for companion plugin. Assuming it is not present!", e)
+            isCompanionPluginAvailable = false
         }
-        if (!groovyPlugins.companionPluginAvailable) {
-            groovyPlugins.companionPluginAvailable = groovyChecker.companionPluginAvailable
+    }
+
+    private fun checkForPlugins(connection: ProjectConnection) {
+        val outputStream = ByteArrayOutputStream()
+        connection.newBuild().forTasks(TASK_NAME).setStandardOutput(outputStream).run()
+        var output = outputStream.toString(StandardCharsets.UTF_8.name())
+        logger.trace("Got output from check task: $output")
+        // remove content until the first { and from behind until the last }
+        output = "{" + output.substringAfter("{").substringBeforeLast("}") + "}"
+
+        // Deserialize the JSON string into a Kotlin object
+        pluginCheckData = Json.decodeFromString<PluginCheckData>(output)
+        logger.debug("Got plugin check data: $pluginCheckData")
+    }
+
+    private fun findTaskRecursively(project: GradleProject): Boolean {
+        // Check if the current project contains the task
+        val taskExistsInCurrentProject = project.tasks.any { it.name == TASK_NAME }
+
+        if (taskExistsInCurrentProject) {
+            return true
         }
+        // Recursively check subprojects
+        return project.children.any { findTaskRecursively(it) }
     }
 
-    private fun getInputStreamFromVirtualFile(virtualFile: VirtualFile): InputStream? {
-        val document = FileDocumentManager.getInstance().getDocument(virtualFile)
-        return document?.let {
-            ByteArrayInputStream(it.text.toByteArray())
+    private fun getBuildFileName(): String {
+        return buildFiles.firstOrNull()?.name ?: "Unknown build file"
+    }
+
+    private fun getCompanionPluginString(): String {
+        val buildFile = getBuildFileName()
+        return if (buildFile.endsWith(KOTLIN_BUILD_FILE)) {
+            KOTLIN_OVERRIDE_PLUGIN
+        } else if (buildFile.endsWith(GROOVY_BUILD_FILE)) {
+            GROOVY_OVERRIDE_PLUGIN
+        } else {
+            ""
         }
     }
 
-    private class PluginsAvailability {
-        var pitestPluginAvailable: Boolean = false
-        var companionPluginAvailable: Boolean = false
+    private fun getPitestPluginName(): List<String> {
+        val isAndroid = pluginCheckData?.androidPitestPluginApplied ?: false
```

Comment:
```suggestion
        val isAndroid = pluginCheckData?.androidPluginApplied ?: false
```

Reply:
I have applied that in 6a6e78808240e870197b75531f527903bdff2f82
I also added some more doc string in the pluginchecker in an effort to make things clearer